packages/blade/codemods/brand-refresh/transformers/index.ts

```diff
@@ -0,0 +1,161 @@
+          return originalString;
+        }
+        return `${fontTokenPrefix}[${fontSizeMap[token]}]`;
+      },
+    )
+    .replace(
+      // gets both .50 and ['50'] or ["50"]
+      /theme\.typography\.lineHeights\.?((\w+)|(\W.*\]))/g,
+      (originalString, match) => {
+        const token = match.replace(/\[|\]|'|"/g, '');
+        const replacement = lineHeightMap[token];
+        if (!replacement) {
+          return originalString;
+        }
+        return `${lineHeightTokenPrefix}[${lineHeightMap[token]}]`;
+      },
+    );
+
+  const j = api.jscodeshift;
+  const root = j.withParser('tsx')(newSource);
+
+  // Select Typography elements based on their names
+  const typographyJSXElements = root
+    .find(j.JSXElement)
+    .filter((path) =>
+      ['Text', 'Title', 'Code', 'Display', 'Heading'].includes(path.value.openingElement.name.name),
+    );
+
+  // Replace Title with Heading and update the 'size' attribute
+  // <Title size="medium"> to <Heading size="xlarge">
+  typographyJSXElements
+    .filter((path) => path.value.openingElement.name.name === 'Title')
+    // replace with Heading
+    .replaceWith((path) => {
+      const { node } = path;
+
+      node.openingElement.name.name = 'Heading';
+      node.closingElement.name.name = 'Heading';
+
+      return node;
+    })
+    .find(j.JSXAttribute)
+    .filter((path) => path.node.name.name === 'size')
+    .replaceWith((path) => {
+      if (isExpression(path.node)) return path.node;
+
+      path.node.value.value = titleToHeadingMap[path.node.value.value] || 'large';
+
+      return path.node;
+    });
+
+  // Remove/Update the Title import from "@razorpay/blade/components"
+  root
+    .find(j.ImportDeclaration)
+    .filter((path) => path.value.source.value === '@razorpay/blade/components')
+    .find(j.ImportSpecifier)
+    .filter((path) => path.value.imported.name === 'Title')
+    .replaceWith((path) => {
+      // Check if Heading import is already present
+      const isHeadingImportPresent = path.parent.value.specifiers.some(
+        (node) => node.imported.name === 'Heading',
+      );
+      // // If Heading import is not present, update the "Title" import to use "Heading"
+      if (!isHeadingImportPresent) {
+        path.value.imported.name = 'Heading';
+      } else {
+        // If "Heading" import is present, remove the "Title" import
+        path.parent.value.specifiers = path.parent.value.specifiers.filter(
+          (node) => node.imported.name !== 'Title',
+        );
+      }
+
+      return path.node;
+    });
+
+  // Remove `type` prop from Typography Components
+  typographyJSXElements
+    .find(j.JSXAttribute) // Find all JSX props
+    .filter(
+      (path, index, self) =>
+        path.node.name.name === 'type' &&
+        index === self.findIndex((obj) => path.node.start === obj.node.start),
+    ) // Filter by name `type` and remove any duplicates
+    .remove();
+
+  // Remove `variant` prop from Heading Component
```

Comment:
In this case, the `<Heading variant="subheading">` instance will turn to `<Text size="small">` I believe.

You can check the mapping here once - https://www.figma.com/file/5BZsOpNjbUHqgVh850yPBW/%5BResearch%5D-Typography-%26-Spacing-Refresh?type=design&node-id=76-82092&mode=design&t=7K0OPHpiW1oM9b6F-0

Ideally anything we remove should have some equivalent closest neighbour that we should match to otherwise UI can break. We can check with RK once if you're not sure what to match to.

Reply:
Can you also check Heading component mapping in this link ^^. Heading size="medium" for example maps to Heading size="small" now and same with other typography components